src/Core/src/Hosting/MauiHandlersServiceProvider.cs

```diff
@@ -4,7 +4,8 @@ namespace Microsoft.Maui.Hosting
 {
 	class MauiHandlersServiceProvider : MauiServiceProvider, IMauiHandlersServiceProvider
 	{
-		public MauiHandlersServiceProvider(IMauiServiceCollection collection) : base(collection)
+		public MauiHandlersServiceProvider(IMauiServiceCollection collection)
+			: base(collection, false)
```

Comment:
Right now, handlers don't use constructor injection, but his could also be pulled from the builder option if need be. This means if the user opts-in to injection, it won't affect performance at all since it only happens for user objects. Since we know we can get all the services from the `IMauiContext`, we don't need it to work.